components/x-topic-search/src/TopicSearch.jsx

```diff
@@ -33,6 +36,22 @@ const topicSearchActions = withActions(({ minSearchLength = 2, maxSuggestions =
 		}
 	},
 
+	topicFollowed (subjectId, subjectName) {
+		const followedTopicIndex = getFollowedTopicIndex(followedTopics, subjectId);
+
+		if (followedTopicIndex === -1) {
+			followedTopics.push({ name: subjectName, uuid: subjectId })
+		}
```

Comment:
do you need to `return { followedTopics };` at the end of this function and `topicUnfollowed`, so the component will update?